spec/cli/commands/ip_spec.rb

```diff
@@ -4,4 +4,63 @@
 
 describe Ronin::CLI::Commands::Ip do
   include_examples "man_page"
+
+  let(:ipcmd) { Ronin::CLI::Commands::Ip.new}
+  let(:localhost) { "127.0.0.1" }
+
+  describe "-r" do
+    it "will reverse an IPv4 address" do
+      ipcmd.options[:reverse] = true
+      expect{ ipcmd.process_value(localhost) }.to output("1.0.0.127.in-addr.arpa\n").to_stdout
+    end
+  end
+
+  describe "--hex" do
+    it "will convert an IPv4 address to hexadecimal" do
+      ipcmd.options[:hex] = true
+      expect{ ipcmd.process_value(localhost) }.to output("0x7f000001\n").to_stdout
+    end
+  end
+
+  describe "--decimal" do
+    it "will convert an IPv4 address to decimal" do
+      ipcmd.options[:decimal] = true
+      expect{ ipcmd.process_value(localhost) }.to output("2130706433\n").to_stdout
+    end
+  end
+
+  describe "--octal" do
+    it "will convert an IPv4 address to octal" do
+      ipcmd.options[:octal] = true
+      expect{ ipcmd.process_value(localhost) }.to output("017700000001\n").to_stdout
+    end
+  end
+
+  describe "--binary" do
+    it "will convert an IPv4 address to binary" do
+      ipcmd.options[:binary] = true
+      expect{ ipcmd.process_value(localhost) }.to output("1111111000000000000000000000001\n").to_stdout
+    end
+  end
+
+  describe "--octal-octet" do
+    it "will convert an IPv4 address to octal by octet" do
+      ipcmd.options[:octal_octet] = true
+      expect{ ipcmd.process_value(localhost) }.to output("0177.00.00.01\n").to_stdout
+    end
+  end
+
+  describe "--hex-octet" do
+    it "will convert an IPv4 address to hex by octet" do
+      ipcmd.options[:hex_octet] = true
+      expect{ ipcmd.process_value(localhost) }.to output("7f.0.0.1\n").to_stdout
+    end
+  end
+
+  describe "--ipv6-compat" do
+    it "will convert an IPv4 address to an IPv6 compatible address" do
+      ipcmd.options[:ipv6_compat] = true
+      expect{ ipcmd.process_value(localhost) }.to output("::ffff:127.0.0.1\n").to_stdout
+    end
+  end
```

Comment:
If you want, you could probably copy/edit these specs to create specs for `#format_ip` as well. Just replace `expect { ... }.to output(...).to_stdout` to `expect(subject.format_ip(ip)).to eq(...)`.

Reply:
That's another thing I did initially but the output of `#format_ip` and `#process_value` differed (I think due to stringy things with `puts`) so I opted for more of an integration test.  Happy to move them back to `#format_ip` if that is preferred.

Reply:
If possible it's always better to have both unit-level and integration-level tests. I'd assume the `#format_ip` outputs should be roughly the same as `#process_value`, but without the new-line characters.

Reply:
I couldn't agree more.  Do you prefer one assertion per test or would you like separate tests broken down by method?

Reply:
I tend to use `describe "#method"` then `context "when bla bla bla is set"` within the `describe` block, as a way to compartmentalize the tests and any `before`/`subject` logic per-test.

Reply:
Example: https://github.com/ronin-rb/ronin/blob/8b84995308814aff6474768a4862dd4766885b52/spec/cli/commands/http_spec.rb#L50-L83